sample-resource-plugin/src/integrationTest/java/org/opensearch/sample/SampleResourcePluginLimitedPermissionsTests.java

```diff
@@ -0,0 +1,456 @@
+import static org.opensearch.test.framework.TestSecurityConfig.AuthcDomain.AUTHC_HTTPBASIC_INTERNAL;
+import static org.opensearch.test.framework.TestSecurityConfig.User.USER_ADMIN;
+
+/**
+ * These tests run with resource sharing enabled and system index protection enabled
+ */
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class SampleResourcePluginLimitedPermissionsTests {
+
+    ResourcePluginInfo resourcePluginInfo;
+
+    @ClassRule
+    public static LocalCluster cluster = new LocalCluster.Builder().clusterManager(ClusterManager.SINGLENODE)
+        .plugin(SampleResourcePlugin.class, PainlessModulePlugin.class)
+        .anonymousAuth(true)
+        .authc(AUTHC_HTTPBASIC_INTERNAL)
+        .users(USER_ADMIN, SHARED_WITH_USER_LIMITED_PERMISSIONS)
+        .nodeSettings(Map.of(SECURITY_SYSTEM_INDICES_ENABLED_KEY, true, OPENSEARCH_RESOURCE_SHARING_ENABLED, true))
+        .build();
+
+    @Before
+    public void setup() {
+        resourcePluginInfo = cluster.nodes().getFirst().getInjectable(ResourcePluginInfo.class);
+    }
+
+    @After
+    public void clearIndices() {
+        try (TestRestClient client = cluster.getRestClient(cluster.getAdminCertificate())) {
+            client.delete(RESOURCE_INDEX_NAME);
+            client.delete(OPENSEARCH_RESOURCE_SHARING_INDEX);
+            resourcePluginInfo.getResourceIndicesMutable().remove(RESOURCE_INDEX_NAME);
+            resourcePluginInfo.getResourceProvidersMutable().remove(RESOURCE_INDEX_NAME);
+        }
+    }
+
+    @Test
+    public void testPluginInstalledCorrectly() {
+        try (TestRestClient client = cluster.getRestClient(USER_ADMIN)) {
+            TestRestClient.HttpResponse pluginsResponse = client.get("_cat/plugins");
+            assertThat(pluginsResponse.getBody(), containsString("org.opensearch.security.OpenSearchSecurityPlugin"));
+            assertThat(pluginsResponse.getBody(), containsString("org.opensearch.sample.SampleResourcePlugin"));
+        }
+    }
+
+    @Test
+    public void testCreateUpdateDeleteSampleResource() throws Exception {
+        String resourceId;
+        String resourceSharingDocId;
+        // create sample resource
+        try (TestRestClient client = cluster.getRestClient(USER_ADMIN)) {
+            String sampleResource = """
+                {"name":"sample"}
+                """;
+
+            TestRestClient.HttpResponse response = client.putJson(SAMPLE_RESOURCE_CREATE_ENDPOINT, sampleResource);
+            response.assertStatusCode(HttpStatus.SC_OK);
+
+            resourceId = response.getTextFromJsonBody("/message").split(":")[1].trim();
+        }
+
+        // Create an entry in resource-sharing index
+        try (TestRestClient client = cluster.getRestClient(cluster.getAdminCertificate())) {
+            // Since test framework doesn't yet allow loading ex tensions we need to create a resource sharing entry manually
```

Comment:
Did we consider finished https://github.com/opensearch-project/OpenSearch/pull/16908 to get this officially supported?

If its officially supported will that obviate the need for the block below:

```
// Also update the in-memory map and get
resourcePluginInfo.getResourceIndicesMutable().add(RESOURCE_INDEX_NAME);
ResourceProvider provider = new ResourceProvider(SampleResource.class.getCanonicalName(), RESOURCE_INDEX_NAME);
resourcePluginInfo.getResourceProvidersMutable().put(RESOURCE_INDEX_NAME, provider);

ResourceSharingClientAccessor.getInstance().setResourceSharingClient(createResourceAccessControlClient(cluster));
```